### ISSUE TITLE
How to Extend Stores and Actions to another stores and actions

### ISSUE THREAD

**Issue by spyalert01** [goatslacker/alt#444]
Hi,
I have a Standard Store and Standard Action.
I would like to use write another store that will extend the Standard Store, and another actions to extend the standard Actions. 
How should I go about doing this?

Example: 
/* standard store */
import Alt from '../../../altControl.js';  
import {createStore, bind} from 'alt/utils/decorators';  
import ActDummy from '../actions/actDummy.js';

@createStore(Alt)
class StoDummyStd {  
	constructor(props){
	}
	
  name = 'awesome';

  @bind(ActDummy.updateName)
  updateName(name) {
    this.name = name;
  }
}
export default StoDummyStd;  

/* new  store that need to extend standard store*/
import Alt from '../../../altControl.js';  
import {createStore, bind} from 'alt/utils/decorators';  
import ActDummy from '../actions/actDummy.js';
import StoDummyStd from './stoDummyStd.js';
@createStore(Alt)
class StoDummy extends StoDummyStd{  
	constructor(props){
		super(props);
	}
  @bind(ActDummy.updateName)
  updateName(name) {
    this.name = name;
	this.nick = 'nick'
  }
}
export default StoDummy;  

It just can't work like this.

    throw new TypeError("Super expression must either be null or a function, n
          ^
TypeError: Super expression must either be null or a function, not object

**Comment by MPiccinato** [comment]
Also curious about this. I have a set of actions available for a frontend app and wanted to extend it for a backend app since it contains all the frontend actions and just needs a few more actions tacked on.

**Comment by MPiccinato** [comment]
I am not really sure if this is the best way to do this...but you can extend the store by creating a new store and passing the existing store in as the second param to alt.CreateActions

```
export default alt.createActions(MyActions, MyActionsToExtend);
```

Where 'MyActionsToExtend' is really the original 'MyActions' class just imported with 'ToExtend' appended.

**Comment by spyalert01** [comment]
Nope. When we use Class to Extend another class, the ChildClass can have the same function Name like the Parent Class, but different set of command. and it work perfectly fine.
in the way you propose, we cannot have the same function name.

**Comment by MPiccinato** [comment]
Revisiting this a bit today as I now have a similar setup as above. When alt goes to generate the actions I am not getting the error you show but instead there are just no methods available in the class 'StoDummy'.  The methods don't exist on the class declaration until I actually instantiate a new StdoDummy() instance.
